airflow/utils/json.py

```diff
@@ -189,7 +189,7 @@ def __init__(self, *args, **kwargs) -> None:
         if not kwargs.get("object_hook"):
             kwargs["object_hook"] = self.object_hook
 
-        patterns = conf.getjson("core", "allowed_deserialization_classes")
+        patterns = cast(list, conf.getjson("core", "allowed_deserialization_classes"))
```

Comment:
This would result in a confusing error if the value is not set to a list. It’s probably better to explicitly check the value is a list instead (and raise a clear message explaining the config value is exact source of failure).

Reply:
Mmm yes, I'd prefer that check at configure time rather than here. The config file shouldn't validate if this is not a list.

Reply:
We can always do both so the user sees the error as quickly as possible, and still guarantee integrity is a faulty value _somehow_ gets passed in here.

Reply:
Well so much for type safety then :/. There is no validation in `airflow.configuration` for these kind of values, so I'll implement it here.

Reply:
You can add a custom validation function like here https://github.com/apache/airflow/blob/3fef6a47834b89b99523db6d97d6aa530657a008/airflow/configuration.py#L349-L351